src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs

```diff
@@ -258,52 +298,106 @@ private static X509Certificate2 GetRootCertificate(X509Chain chain)
             {
+                if (clientCert != null && clientCert.Thumbprint.Equals(thumbprint))
+                {
+                    var clientCertDef = ReadCertificateFromCluster(ctx, Constants.Certificates.Prefix + thumbprint);
+                    throw new InvalidOperationException($"Cannot delete {clientCertDef?.Name} becuase it's the current client certificate being used");
+                }
+
+                if (clientCert != null && Server.ClusterCertificateHolder.Certificate.Thumbprint.Equals(thumbprint))
+                {
+                    var serverCertDef = ReadCertificateFromCluster(ctx, Constants.Certificates.Prefix + thumbprint);
+                    throw new InvalidOperationException($"Cannot delete {serverCertDef?.Name} becuase it's the current server certificate being used");
+                }
+
                 var key = Constants.Certificates.Prefix + thumbprint;
-                using (ctx.OpenWriteTransaction())
+                var definition = ReadCertificateFromCluster(ctx, key);
+                if (definition.SecurityClearance == SecurityClearance.ClusterAdmin || definition.SecurityClearance == SecurityClearance.ClusterNode 
+                    && IsClusterAdmin() == false)
                 {
-                    var certificate = ServerStore.Cluster.Read(ctx, key);
+                    var clientCertDef = ReadCertificateFromCluster(ctx, Constants.Certificates.Prefix + clientCert?.Thumbprint);
+                    throw new InvalidOperationException(
+                        $"Cannot delete the certificate '{definition.Name}' with '{definition.SecurityClearance}' security clearance because the current client certificate being used has a lower clearance: {clientCertDef.SecurityClearance}");
+                }
 
-                    var definition = JsonDeserializationServer.CertificateDefinition(certificate);
+                if (string.IsNullOrEmpty(definition.CollectionPrimaryKey) == false)
+                    throw new InvalidOperationException(
+                        $"Cannot delete the certificate '{definition.Name}' with thumbprint '{definition.Thumbprint}'. You need to delete the primary certificate of the collection: {definition.CollectionPrimaryKey}");
 
-                    if (definition?.SecurityClearance == SecurityClearance.ClusterAdmin && IsClusterAdmin() == false)
-                        throw new InvalidOperationException(
-                            $"Cannot delete the certificate '{definition?.Name}' with 'Cluster Admin' security clearance because the current client certificate being used has a lower clearance: {clientCert}");
+                await DeleteInternal(key);
 
-                    ServerStore.Cluster.DeleteLocalState(ctx, key);
+                foreach (var secondaryKey in definition.CollectionSecondaryKeys) // If it's a collection, also delete the secondary certificates
```

Comment:
Can we make this into a single command that take a collection, this way it will be atomic

Reply:
yes